apps/remote_control/lib/lexical/remote_control/code_intelligence/entity.ex

```diff
@@ -0,0 +1,151 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.Entity do
```

Comment:
I copied this module from the server app.

The reason we put it on the server in the first place was because of `Document.Store`, but now it's time to move it over, which will make it easier for us to test, and a lot of remote control functionality will depend on it as well.

I'll be happy to move relative tests to `remote_control` app after @zachallaun 's PR merging.

Reply:
Sounds good! Agreed that it makes sense in remote control.